src/main/java/com/amazon/opendistroforelasticsearch/security/compliance/ComplianceConfig.java

```diff
@@ -255,6 +255,10 @@ public boolean shouldLogDiffsForWrite() {
         return !shouldLogWriteMetadataOnly() && logDiffsForWrite;
     }
 
+    public boolean shouldLogDiffsForWriteInIndex(final String indexName) {
```

Comment:
nit: introduce `private static boolean isLoggingWriteDiffEnabled(final ComplianceConfig complianceConfig, final String indexName)` to `ComplianceIndexingOperationListenerImpl`.
```suggestion
    private static boolean isLoggingWriteDiffEnabled(final ComplianceConfig complianceConfig, final String indexName) {
        return complianceConfig != null && complianceConfig.shouldLogDiffsForWrite() && complianceConfig.writeHistoryEnabledForIndex(indexName);
    }
```